napari/_qt/layers/_tests/test_qt_image_base_layer_.py

```diff
@@ -94,3 +94,15 @@ def test_clim_slider_step_size_and_precision(qtbot, mag):
     # the slider step size should also be inversely proportional to the data
     # range, with 1000 steps across the data range
     assert np.ceil(popup.slider._step * 10 ** (mag + 4)) == 10
+
+
+@pytest.mark.skip(reason="broken test that needs to be fixed")
+def test_complex_colormap_mode_in_3d(viewermodel_factory):
+    """Test that complex rending in one of the RGB modes works in 3d views."""
+    view, viewer = viewermodel_factory()
+
+    data = np.random.random((10, 10)) + np.random.random((10, 10)) * 1j
+    layer = viewer.add_image(data)
+
+    layer.complex_rendering = 'colormap'
+    viewer.dims.ndisplay = 3
```

Comment:
this line causes a problem in `_set_view_slice` here:
https://github.com/napari/napari/blob/bd538488c9a6f9d6721cec8268610a1ea5b0c7d0/napari/layers/image/image.py#L491

it has to do with hijacking the RGB system for complex numbers... but haven't figured it out yet.